### PR TITLE
Inventory ckan 2.8 builded datapusher

### PR DESCRIPTION
**Pull request by avdata99**
Using a built image of datapusher.
Added a test to check the service running.
I must [fix CKAN](https://github.com/ckan/ckan/pull/5281/files) because [it fails](https://github.com/ckan/ckan/issues/4925) to run datapusher in docker-compose. We will need to port this fix to CKAN 2.8.3. Maybe in a new issue.

### CODE REVIEW

**Comment by avdata99** [comment]
I'm not able to add reviewrs here. @pjsharpe07 @adborden 
I must [fix CKAN](https://github.com/ckan/ckan/pull/5281/files) because [it fails](https://github.com/ckan/ckan/issues/4925) to run datapusher in docker-compose. We will need to port this fix to CKAN 2.8.3. Maybe in a new issue.

**Comment by adborden** [comment]
@avdata99 is this ready for review? It's still labeled WIP?

**Comment by avdata99** [comment]
> @avdata99 is this ready for review? It's still labeled WIP?

I just fix the PR title. It's ready to analyze and merge

**Comment by jbrown-xentity** [comment]
@avdata99 so looking at the CKAN changes, it's making [changes to the datapusher](https://github.com/ckan/ckan/commit/d834f1761b8262b76087bd31d0a2b8f97b8c1e5f), right? Isn't the datapusher integrated into CKAN core at this point?
Also, your requirements file doesn't pull from your fix, it's commented out. Is this the correct behavior?

File: config/production.ini [hunk 1]
```diff
@@ -102,7 +103,7 @@ solr_url = http://solr:8983/solr/inventory
 # original plugins
 # ckan.plugins = usmetadata datajson datastore datapusher stats text_view recline_view googleanalyticsbasic
 
-ckan.plugins = googleanalyticsbasic datastore usmetadata datajson
+ckan.plugins = googleanalyticsbasic datastore datapusher
```
Comment (adborden) [inline config/production.ini]: Why the removal of usmetadata and datajson?

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,10 @@
 -e  git+https://github.com/ckan/ckan.git@ckan-2.8.3#egg=ckan
+
+# datapusher hotfix: CKAN has an error for running with docker. 
+# Fixed in master, need to port to CKAN 2.8.3
+# https://github.com/ckan/ckan/pull/5281/files
+# -e git+https://github.com/avdata99/ckan@my_2.8.3#egg=ckan
```
Comment (adborden) [inline requirements.txt]: Is your CKAN fix _required_ before this PR is merged? Can you please push this to the GSA repo as a new branch `datagov-inventory-2.8`? I just added you as a collaborator https://github.com/GSA/ckan/invitations

File: tests/test.bats [hunk 3]
```diff
@@ -104,3 +104,13 @@ function test_datastore_request () {
     return 1;
   fi
 }
+
+@test "datapusher working" {
+  datapusher_status=$(curl -X GET "http://datapusher:8000/status" | grep -o "push_to_datastore")
```
Comment (adborden) [inline tests/test.bats]: Please add the `--fail` flag so a non 200 status code results in curl failing. Also, no need to `grep` and capture the output, `grep` will return non-zero and fail the test if it doesn't find the pattern. No need for the `if`/`else`. The test is simply:

```suggestion
  curl --silent --fail "http://datapusher:8000/status" | grep push_to_datastore
```

**Comment by adborden** [comment]
Hmm, @avdata99 FWIW, https://github.com/ckan/ckan/commit/d834f1761b8262b76087bd31d0a2b8f97b8c1e5f looks like the wrong fix. If datapusher is hosted on a different Host, shouldn't we just set the `ckan.datapusher.callback_url_base` config?

Your fix changes site_url, which feels wrong to me. site_url should always be the CKAN URL (and `ckan.site_url`.

**Comment by avdata99** [comment]
> Hmm, @avdata99 FWIW, [ckan/ckan@d834f17](https://github.com/ckan/ckan/commit/d834f1761b8262b76087bd31d0a2b8f97b8c1e5f) looks like the wrong fix. If datapusher is hosted on a different Host, shouldn't we just set the `ckan.datapusher.callback_url_base` config?
> 
> Your fix changes site_url, which feels wrong to me. site_url should always be the CKAN URL (and `ckan.site_url`).

@adborden From the point of view of the Datapusher (I mean, inside the Datapusher) you need to connecto to the CKAN site URL in a different way when you are usin docker.

localhost:5000 is a common site_url config but fails if you are usin docker. In that case you need ckan:5000 (or app:5000 or something like this).

When CKAN _talks_ to datapusher [sends a ckan_url](https://github.com/ckan/ckan/commit/d834f1761b8262b76087bd31d0a2b8f97b8c1e5f#diff-7b81d8fca5028033fc446f46a76272a7R136) to allow datapusher to send data back to CKAN. Is just some lines bellow my changes.

If we use public URLs for both services (CKAN and Datapusher) this is not a problem. This just [fails when you are inside Docker](https://github.com/ckan/ckan/issues/4925).

**Pull request merged**